django/db/models/base.py

```diff
@@ -1643,9 +1643,39 @@ def _check_ordering(cls):
         # Convert "-field" to "field".
         fields = ((f[1:] if f.startswith('-') else f) for f in fields)
 
-        # Skip ordering in the format field1__field2 (FIXME: checking
-        # this format would be nice, but it's a little fiddly).
-        fields = (f for f in fields if LOOKUP_SEP not in f)
+        # seprate related field and non related fields
+        _fields = []
+        related_fields = []
+        for f in fields:
+            if LOOKUP_SEP in f:
+                related_fields.append(f)
+            else:
+                _fields.append(f)
+        fields = iter(_fields)
+        related_fields = iter(related_fields)
+
+        # check related fields
+        for field in related_fields:
+            parts = field.split(LOOKUP_SEP)
+            _cls = cls
+            fld = None
```

Comment:
This line can be removed :thinking:.

Reply:
This line cannot be removed, because if we will have an error on `fld = _cls._meta.get_field(part)` at the first round of loop. the `fld` variable on `if fld is None or fld.get_transform(part) is None:` line in `except` is `Unbounded` and we will get the `UnboundLocalError`

Reply:
You're right, thanks :+1: